### PR TITLE
Extract HttpCaching

### PR DESCRIPTION
**Pull request by peterhicks**
This PR extracts the HttpCaching concern from the main Rails codebase, and implements tests for it. It introduces a dependency on `flex_commerce_api`, and adds `webmock` to the Gemspec to assist with testing.

### CODE REVIEW

File: bin/console [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env ruby
 
 require "bundler/setup"
-require "shift_commerce/rails"
+require "shiftcommerce-rails"
```
Comment (kmcd) [inline bin/console]: Typo?

Comment (kmcd) [inline bin/console]: Coolio!

File: spec/dummy/config/initializers/flex_commerce_api.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FlexCommerceApi.config do |config|
+  config.flex_root_url ||= 'http://shiftcommmerce-api:3000'
+  config.flex_account = 'testaccount'
+  config.flex_api_key = '314e47ece578a11e6e77c6153229cd25'
```
Comment (ryantownsend) [inline spec/dummy/config/initializers/flex_commerce_api.rb]: I'm presuming this is fake mock data? If not, it should be an ENV var

Comment (peterhicks) [inline spec/dummy/config/initializers/flex_commerce_api.rb]: It's fake, yes – it prevents `flex-ruby-gem` from throwing errors. I may change it to just be "test" or something similar so it's obvious that it's mock data.

**Pull request merged**